src/ppx/application.cpp

```diff
@@ -842,6 +842,16 @@ void Application::InitStandardKnobs()
         "Prints a list of the available GPUs on the current system with their "
         "index and exits (see --gpu).");
 
+    mStandardOpts.pMetricsFilename =
```

Comment:
Can you please also add proper handling if this value is never set, as discussed?

Reply:
Done, I've added checks to `Application::SaveMetricsReportToDisk` to prevent `nullptr`s from crashing the application.